apps/core-api/src/controllers/prm.controller.ts

```
@@ -19,8 +25,13 @@ export const createEntity = async (
 
     const prmService = PrmService[entityType.data];
 
-    const entityDefinition = req.body; // TODO: Validate entity definition
-    const createdEntity = await prmService.create(entityDefinition);
+    const schema = getEntityDefinitionSchema(entityType.data);
+    const entityDefinition = schema.safeParse(req.body);
+    if (entityDefinition.success === false) {
+      res.status(400).json(entityDefinition.error);
+      return;
+    }
+    const createdEntity = await prmService.create(entityDefinition.data);
```

We should validate what we return as well

---

We are using the schema to parse the results from the database in the Store methods. Would you suggest to do it here instead, or in both places?

---

I think the store needs to validate because it is relying on an accurate sql query there and we want to surface that error as soon as possible.

Maybe we can rely on type safety to ensure that it stays valid throughout. I like to be overly cautious but I think the store validation is the minimum needed.